File: src/static/scripts/components/chess/chessConsts.js
const KNIGHT_OFFSET_DIFFS = [[1,2],[1,-2],[-1,2],[-1,-2],[2,1],[2,-1],[-2,1],[-2,-1]];
const BISHOP_OFFSET_DIFFS = [[1,1],[1,-1],[-1,1],[-1,-1]];
const ROOK_OFFSET_DIFFS = [[0,1],[0,-1],[1,0],[-1,0]];

const offsetMap = (diff) => ({row: diff[0], col: diff[1]});

export const KNIGHT_OFFSETS = KNIGHT_OFFSET_DIFFS.map(offsetMap);
export const BISHOP_OFFSETS = BISHOP_OFFSET_DIFFS.map(offsetMap);
export const ROOK_OFFSETS = ROOK_OFFSET_DIFFS.map(offsetMap);
export const ROYAL_OFFSETS = BISHOP_OFFSETS.concat(ROOK_OFFSETS);

export const Colors = Object.freeze({
    WHITE: 'White',
    BLACK: 'Black'
})

export const PieceTypes = Object.freeze({
    PAWN: 'Pawn',
    BISHOP: 'Bishop',
    QUEEN: 'Queen',
    KING: 'King',
    ROOK: 'Rook',
    KNIGHT: 'Knight'
})

File: src/backend/services/chess/chessConsts.py
from enum import Enum

KNIGHT_OFFSETS = [[1, 2], [1, -2], [-1, 2], [-1, -2], [2, 1], [2, -1], [-2, 1], [-2, -1]]
ROOK_OFFSETS = [[0, 1], [0, -1], [1, 0], [-1, 0]]
BISHOP_OFFSETS = [[1, 1], [1, -1], [-1, 1], [-1, -1]]
ROYAL_OFFSETS = [[1, 1], [1, -1], [-1, 1], [-1, -1], [0, 1], [0, -1], [1, 0], [-1, 0]]

wkrStartTile = "77"
wknStartTile = "76"
wkbStartTile = "75"
wkStartTile = "74"
wqStartTile = "73"
wqbStartTile = "72"
wqnStartTile = "71"
wqrStartTile = "70"

bkrStartTile = "07"
bknStartTile = "06"
bkbStartTile = "05"
bkStartTile = "04"
bqStartTile = "03"
bqbStartTile = "02"
bqnStartTile = "01"
bqrStartTile = "00"

ROOK = "Rook"
KING = "King"
QUEEN = "Queen"
BISHOP = "Bishop"
KNIGHT = "Knight"
PAWN = "Pawn"

BLACK = "Black"
WHITE = "White"

class MoveType(Enum):
    CASTLE = "Castle"
    EN_PASSANT = "En Passant"
    NORMAL = "Normal"
    # CAPTURE = "Capture"

class MoveSide(Enum):
    QUEENSIDE = "Queenside"
    KINGSIDE = "Kingside"